MovieApp/Scenes/HomePage/HomeViewController.swift

```diff
@@ -0,0 +1,364 @@
+            ),
+            moviesCollectionView.trailingAnchor.constraint(
+                equalTo: view.trailingAnchor,
+                constant: Constants.MovieCollectionView.trailingPadding
+            ),
+            moviesCollectionView.bottomAnchor.constraint(
+                equalTo: view.bottomAnchor,
+                constant: Constants.MovieCollectionView.bottomPadding
+            )
+        ])
+    }
+    
+    private func setUpNavigationBar() {
+        view.addSubview(navigationBar)
+        navigationBar.delegate = self
+        navigationBar.homeButton.isSelected = true
+        navigationBar.homeButton.backgroundColor = Constants.Color.yellow
+        
+        NSLayoutConstraint.activate([
+            navigationBar.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            navigationBar.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            navigationBar.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        ])
+        navigationBar.translatesAutoresizingMaskIntoConstraints = false
+    }
+    private var selectedIndexPath: IndexPath?
+
+}
+
+//MARK: - Filter Button
+extension HomeViewController: SearchBarDelegate {
+    func filterButtonTap() {
+        categoryCollectionView.isHidden.toggle()
+        
+        if !categoryCollectionView.isHidden {
+            titleLabelTopConstraint.constant = Constants.TitleLabel.newTopPadding
+            movieCollectionViewTopConstraint.constant = Constants.MovieCollectionView.topPadding
+        } else {
+            titleLabelTopConstraint.constant = Constants.TitleLabel.topPadding
+            movieCollectionViewTopConstraint.constant = Constants.MovieCollectionView.topPadding
+        }
+        view.layoutIfNeeded()
+    }
+}
+
+//MARK: - Movie Cell Delegate
+extension HomeViewController: MovieCellDelegate {
+    func movieViewTap() {
+        let detailsViewController = DetailsViewController()
+        navigationController?.pushViewController(detailsViewController, animated: false)
+    }
+}
+
+//MARK: - Navigation Bar Buttons
+extension HomeViewController: NavigationBarDelegate {
+    func homeButtonTap() {
+        let homeViewController = HomeViewController()
+        navigationController?.pushViewController(homeViewController, animated: false)
+    }
+    
+    func favoritesButtonTap() {
+        let favoriteMoviesViewController = FavoriteMoviesViewController()
+        navigationController?.pushViewController(favoriteMoviesViewController, animated: false)
+    }
+}
+
+//MARK: - Collection View DataSource
+extension HomeViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView === categoryCollectionView {
+            return 2
+        } else if collectionView === moviesCollectionView {
+            return 10
+        }
+        return 10
```

Comment:
შეგიძლია ასე დაწერო, ან უბრალოდ ბოლო რეთანრი გადააგდო და პირდაპირ `if else`-ის ნაცვლად `else` დაწერო.

```suggestion
        collectionView === categoryCollectionView ? 2 : 10
```